src/lib/draw/route/RouteControls.svelte

```diff
@@ -22,6 +22,28 @@
   function toggleSnap() {
     $routeTool!.toggleSnapMode();
   }
+
+  function loadNodes(show: boolean) {
+    if (show && $showAllNodesGj.features.length == 0) {
+      $showAllNodesGj = JSON.parse($routeTool!.inner.debugSnappableNodes());
+    }
+  }
+  $: loadNodes($showAllNodes);
+
+  // Show snappable nodes when going from freehand to snapped
+  let first = true;
+  function freeToSnapped(snapped: boolean) {
+    if (first) {
+      first = false;
+      return;
+    }
+    if (snapped) {
+      $showAllNodes = true;
+      // TODO Not sure why, but this isn't triggering the reactive statement above
```

Comment:
is it to do with it being a pointer effectively? is the reactive statement listening to the store itself rather than what's stored inside of it?